src/org/thoughtcrime/securesms/notifications/NotificationCenter.java

```diff
@@ -467,15 +482,39 @@ public void addNotification(int chatId, int msgId) {
             // add notification, we use one notification per chat,
             // esp. older android are not that great at grouping
             notificationManager.notify(ID_MSG_OFFSET + chatId, builder.build());
+
+            // group notifications together in a summary, this is possible since SDK 24 (Android 7)
+            // https://developer.android.com/training/notify-user/group.html
+            // in theory, this won't be needed due to setGroup(), however, in practise, it is needed up to at least Android 10.
+            if (Build.VERSION.SDK_INT >= 24) {
+                NotificationCompat.Builder summary = new NotificationCompat.Builder(context, notificationChannel)
+                        .setGroup(GRP_MSG)
+                        .setGroupSummary(true)
+                        .setSmallIcon(R.drawable.icon_notification)
+                        .setColor(context.getResources().getColor(R.color.delta_primary))
+                        .setCategory(NotificationCompat.CATEGORY_MESSAGE)
+                        .setContentTitle("Delta Chat") // content title would only be used on SDK <24
+                        .setContentText("New messages") // content text would only be used on SDK <24
+                        .setContentIntent(getOpenChatlistIntent());
+                notificationManager.notify(ID_MSG_SUMMARY, summary.build());
+            }
         });
     }
 
     public void removeNotifications(int chatId) {
+        boolean removeSummary = false;
         synchronized (inboxes) {
             inboxes.remove(chatId);
+            removeSummary = inboxes.isEmpty();
         }
-        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
-        notificationManager.cancel(ID_MSG_OFFSET + chatId);
+
+        try {
+            NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
+            notificationManager.cancel(ID_MSG_OFFSET + chatId);
+            if (removeSummary) {
+                notificationManager.cancel(ID_MSG_SUMMARY);
+            }
+        } catch (Exception e) { Log.w(TAG, e); }
     }
 
     public void removeAllNotifiations() {
```

Comment:
this calls NotificationManagerCompat::cancelAll(), i first suspected that this would also remvove the permantent notification, but it actually does not in my tests.